packages/formstr-app/src/containers/ResponsesNew/components/ResponseDetailModal.tsx

```diff
@@ -0,0 +1,180 @@
+    if (isVisible && event) {
+      const authorNpub = nip19.npubEncode(event.pubkey);
+      const timestamp = new Date(event.created_at * 1000).toLocaleString();
+      setMetaData({ author: authorNpub, timestamp });
+      setProcessedData([{ key: 'loading', question: 'Processing...', answer: '...' }]); 
+    } else {
+      setProcessedData([]);
+      setMetaData({});
+    }
+  }, [isVisible, event, formSpec, editKey]); 
+  const getInputs = (responseEvent: Event): Tag[] => {
+     if (responseEvent.content === "") {
+       return responseEvent.tags.filter((tag): tag is Tag => Array.isArray(tag) && tag[0] === "response");
+     } else if (editKey) {
+       try {
+         let conversationKey = nip44.v2.utils.getConversationKey(
+           editKey,
+           responseEvent.pubkey
+         );
+         let decryptedContent = nip44.v2.decrypt(
+           responseEvent.content,
+           conversationKey
+         );
+         const parsed = JSON.parse(decryptedContent);
+         if(Array.isArray(parsed)) {
+             return parsed.filter(
+               (tag: Tag): tag is Tag => Array.isArray(tag) && tag[0] === "response"
+             );
+         }
+         console.warn("Decrypted content is not an array:", parsed);
+         return [];
+       } catch (e) {
+           console.error("Failed to parse decrypted response content in modal:", e);
+           return [];
+       }
+     } else {
+       console.warn("Cannot decrypt response in modal: EditKey not available.");
+       return [];
+     }
+   };
+
+  const processEventForDisplay = (
+    responseEvent: Event,
+    spec: Tag[] | null | undefined,
+    key: string | undefined | null
+  ): ResponseDetailItem[] => {
+    if (!spec) {
+         if (responseEvent.content !== "" && !key) return [{ key: 'error-no-spec-no-key', question: 'Error', answer: 'Cannot display details. Form specification is encrypted, and the required key is missing.' }];
+         return [{ key: 'error-no-spec', question: 'Error', answer: 'Could not load form specification to display question labels.' }];
+    }
+    const inputs = getInputs(responseEvent); 
+    if (inputs.length === 0) {
+         if (responseEvent.content !== "" && !key) return [{ key: 'error-decrypt', question: 'Access Denied', answer: 'Cannot decrypt response content without the correct key.' }];
+         return [{ key: 'no-inputs', question: 'Info', answer: 'No response data found in this event.' }];
+    }
+    const details: ResponseDetailItem[] = inputs.map((inputTag) => {
+      const [_resPlaceholder, fieldId, answerValue, metadataString] = inputTag;
+      const questionField = spec.find(
+        (tag): tag is Field => tag[0] === "field" && tag[1] === fieldId
+      );
+      let questionLabel = `Question ID: ${fieldId}`;
+      let displayAnswer = answerValue ?? "N/A";
+      if (questionField) {
+        questionLabel = questionField[3] || questionLabel;
+        if (questionField[2] === "option" && answerValue) {
```

Comment:
we do this in responses too, let's make a util for it so we dont have redundant code